src/xtopcom/xrpc/xrpc_eth_query_manager.cpp

```diff
@@ -676,10 +677,18 @@ void xrpc_eth_query_manager::eth_estimateGas(xJson::Value & js_req, xJson::Value
         break;
     }
 
-    case evm_common::Revert:
+    case evm_common::Revert: {
+        std::string revert_msg;
+        auto t = evm_common::xabi_decoder_t::build_from_hex_string(extra_msg);
+        auto selector = t.extract<evm_common::xfunction_selector_t>();
+        if (selector.method_id == abi_error) {
+           revert_msg = ": " + t.extract<std::string>();
+        }
         js_rsp["error"]["code"] = eth::enum_eth_rpc_execution_reverted;
-        js_rsp["error"]["message"] = "execution reverted";
+        js_rsp["error"]["data"] = extra_msg;
+        js_rsp["error"]["message"] = "execution reverted"+revert_msg;
```

Comment:
remove variable `revert_msg`, move `js_rsp["error"]["message"] = "execution reverted: " + t.extract<std::string>();` into the body of `if (selector.method_id == abi_error)`